packages/mobx-react/__tests__/observer.test.tsx

```diff
@@ -908,3 +908,46 @@ test("Missing render should throw", () => {
     }
     expect(() => observer(Component)).toThrow()
 })
+
+test("this.context is observable if ComponentName.contextType is set", () => {
+    const Context = createContext({})
+
+    class Parent extends React.Component<any> {
+        state = { counter: 0 }
+
+        componentDidMount() {
+            this.setState({ counter: 1 })
+        }
+
+        render() {
+            return <Context.Provider value={this.state}>{this.props.children}</Context.Provider>
+        }
+    }
+
+    @observer
+    class Child extends React.Component {
+        static contextType = Context
+
+        @computed
+        get counterValue() {
+            return (this.context as any).counter
+        }
+
+        render() {
+            return <div />
```

Comment:
what you want to test here as well is `<div>{this.counterValue}</div>`, is getting rerendered (once) to make sure that the `@observer` as well picks up the change in the computed value, as createObservableProp sets some internal flags, for which I'm not 100% sure that is correct in the case of context (I suspect it is correct)